src/renderer/base/FontCache.h

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+namespace Microsoft::Console::Render::FontCache
+{
+    namespace details
+    {
+        inline const std::vector<wil::com_ptr<IDWriteFontFile>>& getNearbyFontFiles(IDWriteFactory5* factory5)
+        {
+            static const auto fontFiles = [=]() {
+                std::vector<wil::com_ptr<IDWriteFontFile>> files;
+
+                const std::filesystem::path module{ wil::GetModuleFileNameW<std::wstring>(nullptr) };
+                const auto folder{ module.parent_path() };
+
+                for (const auto& p : std::filesystem::directory_iterator(folder))
+                {
+                    if (til::ends_with(p.path().native(), L".ttf"))
+                    {
+                        wil::com_ptr<IDWriteFontFile> fontFile;
+                        if (SUCCEEDED_LOG(factory5->CreateFontFileReference(p.path().c_str(), nullptr, fontFile.addressof())))
```

Comment:
so it's safe to use these objects on different threads because it is a SHARED-type factory? neato

Reply:
`DWRITE_FACTORY_TYPE_SHARED` only enables the use of the Windows Font Cache service for DirectWrite. It's unspecified whether DirectWrite is thread-safe, but it should be, since Direct2D can be used across threads. I'm gonna ask one of the dwrite devs about this...